roles/nova-common/tasks/ceph_integration.yml

```diff
@@ -0,0 +1,53 @@
+---
+- name: fetch rados.py
+  get_url: url={{ ceph.rados_url }}
+           dest="{{ openstack_source.virtualenv_base }}/nova/lib/python2.7/site-packages/rados.py"
+           owner=root
+           group=root
+           mode=0644
+  when: openstack_install_method == 'source'
+
+- name: fetch rbd.py
+  get_url: url={{ ceph.rbd_url }}
+           dest="{{ openstack_source.virtualenv_base }}/nova/lib/python2.7/site-packages/rbd.py"
+           owner=root
+           group=root
+           mode=0644
+  when: openstack_install_method == 'source'
+
+- name: fetch rados.py
+  get_url: url={{ ceph.rados_url }}
+           dest="{{ lookup('items', 'openstack_package.virtualenv_base') }}/nova/lib/python2.7/site-packages/rados.py"
+           owner=root
+           group=root
+           mode=0644
+  when: openstack_install_method == 'package'
+
+- name: fetch rbd.py
+  get_url: url={{ ceph.rbd_url }}
+           dest="{{ lookup('items', 'openstack_package.virtualenv_base') }}/nova/lib/python2.7/site-packages/rbd.py"
+           owner=root
+           group=root
+           mode=0644
+  when: openstack_install_method == 'package'
+
+- name: fetch cinder keyring
+  slurp: path=/etc/ceph/ceph.client.cinder.keyring
+  run_once: true
+  delegate_to: "{{ groups['ceph_monitors'][0] }}"
+  register: cinder_keyring
+
+- name: write cinder keyring
+  copy:
+    dest: "{{ cinder_keyring['source'] }}"
+    content: "{{ cinder_keyring['content'] | b64decode }}"
+    mode: 0644
+    #owner: cinder
+    #group: cinder
```

Comment:
We have referred to the ceph doc. http://docs.ceph.com/docs/master/rbd/rbd-openstack/#setup-ceph-client-authentication where it mentions use of ```cinder.keyring``` after line “Nodes running nova-compute need the keyring file for the nova-compute process:”. Hence we did not find a necessity to have nova keyring separately. However we can add separate nova keyring if Bluebox design suggests so.